### DISCUSSION TITLE
List symbols empty response array

### DISCUSSION THREAD

**Discussion by sansmoraxz** [typescript-language-server/typescript-language-server#886] [Q&A] [answered]
I am trying to use it to get the symbols list for specific document/ workspace. I am getting empty response array through api call with `stdio`.

Is there any special requirements/ do we need to pass in we need to enable through initialization api or with notification call? IMO that should've thrown error.

Not sure what I am missing.

Reference code I am trying with:
https://github.com/sansmoraxz/tambla/blob/f28812b14ccef628c22177ede4c4342ff60b173c/test/lsp.test.ts

With logs:
https://gist.github.com/sansmoraxz/797248a252ade0a6ca16beda9ae164f7/7ac6da9b346ed8cf879efee6412e833862a08c59

**Comment by rchl** [comment, marked as answer]
Generally server requires that the file is opened first. Many methods will return an empty result or null if file is not opened. For example: https://github.com/typescript-language-server/typescript-language-server/blob/caca0071095557ecc43d8695d922a4cf5e38a2ad/src/lsp-server.ts#L465-L469

And note that this path is not a valid URI: https://github.com/sansmoraxz/tambla/blob/f28812b14ccef628c22177ede4c4342ff60b173c/test/lsp.test.ts#L66-L69

**Reply by sansmoraxz** [reply]
Ok so file open / close notifies does it. Weirdly enough [jedi-language-server](https://github.com/pappasam/jedi-language-server) worked without those notifications, was expecting the same here.

Is there any guidelines for LSP clients? Trying to implement on something that's not a text editor and the material seems surprisingly sparse.

**Reply by rchl** [reply]
Has been discussed in the LSP spec repo: https://github.com/microsoft/language-server-protocol/issues/1912
